outbox/src/main/kotlin/com/ritense/outbox/OutboxAutoConfiguration.kt

```diff
@@ -0,0 +1,102 @@
+ */
+
+package com.ritense.outbox
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.ritense.outbox.publisher.LoggingMessagePublisher
+import com.ritense.outbox.publisher.MessagePublisher
+import com.ritense.outbox.publisher.PollingPublisherJob
+import com.ritense.outbox.publisher.PollingPublisherService
+import org.springframework.boot.autoconfigure.AutoConfigureAfter
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
+import org.springframework.boot.autoconfigure.domain.EntityScan
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration
+import org.springframework.boot.autoconfigure.liquibase.LiquibaseProperties
+import org.springframework.boot.autoconfigure.orm.jpa.HibernateJpaAutoConfiguration
+import org.springframework.boot.context.properties.EnableConfigurationProperties
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.core.Ordered.LOWEST_PRECEDENCE
+import org.springframework.core.annotation.Order
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories
+import org.springframework.transaction.PlatformTransactionManager
+import javax.sql.DataSource
+
+@Configuration
+@EnableJpaRepositories(
+    basePackageClasses = [
+        OutboxMessageRepository::class
+    ]
+)
+@EntityScan(basePackages = ["com.ritense.outbox"])
+@AutoConfigureAfter(DataSourceAutoConfiguration::class, HibernateJpaAutoConfiguration::class)
+@EnableConfigurationProperties(LiquibaseProperties::class)
+class OutboxAutoConfiguration {
+
+    @Bean
+    @ConditionalOnMissingBean(OutboxLiquibaseRunner::class)
+    fun outboxLiquibaseRunner(
+        liquibaseProperties: LiquibaseProperties,
+        datasource: DataSource
+    ): OutboxLiquibaseRunner {
+        return OutboxLiquibaseRunner(liquibaseProperties, datasource)
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(OutboxService::class)
+    fun outboxService(
+        outboxMessageRepository: OutboxMessageRepository,
+        objectMapper: ObjectMapper,
+    ): OutboxService {
+        return OutboxService(
+            outboxMessageRepository,
+            objectMapper
+        )
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(PollingPublisherService::class)
+    fun pollingPublisherService(
+        outboxService: OutboxService,
+        messagePublisher: MessagePublisher,
+        platformTransactionManager: PlatformTransactionManager
+    ): PollingPublisherService {
+        return PollingPublisherService(
+            outboxService,
+            messagePublisher,
+            platformTransactionManager
+        )
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(PollingPublisherJob::class)
+    fun pollingPublisherJob(
+        pollingPublisherService: PollingPublisherService
+    ): PollingPublisherJob {
+        return PollingPublisherJob(pollingPublisherService)
+    }
+
+    @Order(LOWEST_PRECEDENCE)
```

Comment:
Why the lowest?

Reply:
So that any other MessagePublisher will get a higher priority

Reply:
I doesn't look like we support more than one instance though.